Compiler/src/tfuncs.jl

```diff
@@ -1210,7 +1210,7 @@ end
     nf = length(ftypes)
     # If no value has this type, then this statement should be unreachable.
     # Bail quickly now.
-    if !has_concrete_subtype(s) || nf == 0
+    if !invokelatest(has_concrete_subtype, s)::Bool || nf == 0
```

Comment:
It be good to know why you are getting the wrong answer here, but we could at least change this to use an explicit `compilerbarrier` instead of the vague `invokelatest`

Reply:
Fixed in https://github.com/JuliaLang/julia/pull/57764/commits/f026e6ed809fc7421dbfdfde72e9837bde3fe865, we were incorrectly annotating `jl_datatype_t->flags` as `const` for TBAA.